lib/github_bitbucket_deployer/git.rb

```diff
@@ -5,23 +5,30 @@
 
 module GithubBitbucketDeployer
   class Git
-    attr_reader :bitbucket_repo_url, :git_repo_name, :id_rsa, :repo_dir, :logger, :force, :force_pristine_repo_dir
+    attr_reader :bitbucket_repo_url, :git_repo_name, :id_rsa, :repo_dir, :logger,
+                :force, :force_pristine_repo_dir
 
-    def initialize(options)
-      @bitbucket_repo_url      = options[:bitbucket_repo_url]
-      @git_repo_name           = options[:git_repo_name]
-      @id_rsa                  = options[:id_rsa]
-      @logger                  = options[:logger]
-      @repo_dir                = options[:repo_dir]
-      @force                   = options.fetch(:force, true)
-      @force_pristine_repo_dir = options.fetch(:force_pristine_repo_dir, false)
+    def initialize(params)
+      params.keys.each do |key|
+        instance_variable_set("@#{key}", params[key])
+      end
+      @force = true if force.nil?
     end
 
     def push_app_to_bitbucket(remote = 'bitbucket', branch = 'master')
-      logger.info('push_app_to_bitbucket')
-      add_remote(remote)
-      with_ssh { yield(repo) } if block_given?
-      push(remote, branch)
+      begin
+        logger.info('push_app_to_bitbucket')
+        add_remote(remote)
+        with_ssh { yield(repo) } if block_given?
+        push(remote, branch)
+      rescue => e
+        if force_pristine_repo_dir && !@already_forced_pristine_repo_dir
+          make_repo_dir_pristine
+          retry
```

Comment:
If we encounter an error, delete the repo-dir and `retry` one time

Reply:
Literally never heard of this keyword.

Reply:
https://github.com/g5search/github_bitbucket_deployer/blob/master/github_bitbucket_deployer.gemspec#L20